peps/pep-0776.rst

```diff
@@ -0,0 +1,794 @@
+CI Resources
+------------
+
+Pyodide can be built and tested on any Linux with a reasonably recent version of
+Node.js. Anaconda has offered to provide physical hardware to run Emscripten
+buildbots, maintained by Russell Keith-Magee.
+
+CPython does not currently test Tier 3 platforms on GitHub Actions, but if this
+ever changes, their Linux runners are able to build and test Emscripten Python.
+
+Packaging
+---------
+
+Existing Package Support
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pyodide currently maintains ports of 255 different packages at the time of this
+writing, including major scientific Python packages like NumPy, SciPy, pandas,
+Polars, scikit-learn, OpenCV, PyArrow, and Pillow as well as general purpose
+packages like aiohttp, Requests, Pydantic, cryptography, and orjson.
+
+About 60 packages are also testing against Pyodide in their CI, including NumPy,
+pandas, awkward-cpp, scikit-image, statsmodels, PyArrow, Hypothesis, and PyO3.
+
+Emscripten Wheel Format
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Emscripten wheels will use either the format ``emscripten_<version>_wasm32`` or
+``pyodide_<abi>_wasm32``. For example:
+
+* ``emscripten_3_1_58_wasm32``
+* ``pyodide_2025_0_wasm32``
+
+The first triple is ambiguous, since even with Emscripten 3.1.58 it is possible
+to link dynamic libraries that require a large number of distinct ABIs,
+depending on linker and compiler options. It is our intent that the
+``pyodide_2025_0`` specifies the particular ABI. Thus, the relationship between
+``pyodide_<abi>`` and ``emscripten_<version>`` is intended to be the same as the
+relationship between ``manylinux<version>`` and ``linux``.
+
+The specification of the ``pyodide_<abi>`` ABI includes:
+
+* Which version of the Emscripten compiler is used
+* What libraries are statically linked with the interpreter
+* What stack unwinding ABI is to be used
+* Which runtime platform features are required to be present
+
+and a handful of other similar details that affect the ABI.
+
+The ABI is selected by choosing the appropriate version of the Emscripten
+compiler and passing appropriate compiler and linker flags. It is possible for
+other people to build their own Python interpreter that is compatible with the
+Pyodide ABI, it is not necessary to use the Pyodide distribution itself.
+
+The ``pyodide build`` tool knows how to create wheels that match our ABI. As an
+alternative,
+`the auditwheel-emscripten tool <https://github.com/ryanking13/auditwheel-emscripten>`__
+is capable of performing basic compatibility checks, vendoring shared libraries,
+and retagging the wheel from ``emscripten_<version>`` to ``pyodide_<abi>``. Unlike
+with manylinux, there is no need for a Docker container to build the
+``pyodide_<abi>`` wheels. All that is needed is a Linux machine and appropriate
+versions of Python, Node.js, and Emscripten.
+
+
+PEP 11
+------
+
+:pep:`11` will be updated to indicate that Emscripten is supported, specifically
+the triples ``wasm32-unknown-emscripten_xx_xx_xx``.
```

Comment:
To confirm - are the wildcards going to be part of the PEP 11 spec, or will be updating PEP11 each time a new specific emscripten version is added to the test pool?

Reply:
I was hoping to keep the wild cards. Maybe I should give a concrete example here or otherwise clarify that they are standing in for a version number. Also, we don't support _all_ emscripten versions, so we should probably add where we plan to indicate this.